### PR TITLE
Armor tech of the current* year | Protect yourself from all them "guncargos" the old people keep warning you about

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

We have the guns, we have the bullets, now what? People slaughter eachother in fast fights, any why? Nobody's got protection like they ought to have!

So, lets make some ballistic armor for the real shootouts.

image

I'll give you a little rundown, but there's a few common things between these armors we can get out of the way now.

- Every one of these has practically no protection against lasers, their specialization is bullets and bullets alone, well a little bit of melee I guess but mostly bullets.
- Every bit of helmet and vest comes as sets of things with similar stats, even if the names are different. Really the only time you have to worry about this is the first set of armor where the helmet has a different name than the vest does.

Lets start with the simplest of all the vests, the "soft" vest and the ballistic helmet.

image

With both pieces of armor having 50 bullet armor, what's the deal? The deal, my friend, is 30 wound armor. They may not protect you the best, but they sure will stop you from bleeding if you get shot where the armor is.

The hardened armors.

image

These bad boys also sport 50 bullet armor, a lot less wound protection, so why bother with these? They have a special ability, they remove the armor penetration of any bullet that hits them. That's right, a bullet could have 100 armor penetration and it wouldn't matter, its gone! Down to zero, now it'll have to play nice with the bullet armor like god intended.

And lastly, the sacrificial armor.

image

Quite the name isn't it? Well, it does exactly as advertised. With a massive 80 bullet armor, seems like quite the steal of a deal doesn't it? Here's the catch, this armor can get damaged by gunfire. That's right, you get shot in the armor enough, it'll protect you sure, but it might just break and fall right off after enough.

## How This Contributes To The Skyrat Roleplay Experience

We've got a lot of cheap guns, but not a whole lot of cheap ways to really protect yourself from them. This is just part one of a series of prs to make protection a bit more common for the common man.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image
image

</details>

## Changelog

:cl:
add: A whole host of new armors for protecting yourself from being turned into red mist by a psycho with a rifle are for sale at relatively low cost from Sol Defense in cargo!
image: Sprites for every piece of new armor done by yours truly.
/:cl:

### CODE REVIEW

**Comment by projectkepler-ru** [comment]
armor that break?
the hell are we, CM?

**Comment by SpessManCN** [comment]
dream come true

**Comment by Plum-but-gayer** [comment]
'sacrificial' peacekeeper armor is so soooo flavorful for Lopland. looks gr8, also gives seccies and others something to buy aside from guns.

File: modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm [hunk 1]
```diff
@@ -5,17 +5,38 @@
 // Basic armor vests
 
 /datum/armament_entry/company_import/sol_defense/armor
-	subcategory = "Light Body Armor"
-
-/datum/armament_entry/company_import/sol_defense/armor/slim_vest
-	name = "type I vest - slim"
```
Comment (RatFromTheJungle) [inline modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm:11]: as quirky as the new armors are, you shouldn't remove this until there's another option to get standard armor that isn't 'through the crates you can't buy without sec access'

**Comment by W4rd3nn** [comment]
do these come with jumpsuits? if not, wat are da jumpsuits ur using?

File: modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm (hunk 1 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm:11]: If you’re looking for “standard” anything, company imports will soon be not the place to do that once I am done with it, hope this clears things up

**Comment by Paxilmaniac** [comment]
> do these come with jumpsuits? if not, wat are da jumpsuits ur using?

any of the screenshots with the blue background are using jumpsuits that only exist in aseprite for now

**Comment by W4rd3nn** [comment]
> > do these come with jumpsuits? if not, wat are da jumpsuits ur using?
> 
> any of the screenshots with the blue background are using jumpsuits that only exist in aseprite for now

*grips table really hard*

gimme,.,....... drip.....

**Comment by ChickyRooster** [comment]
> any of the screenshots with the blue background are using jumpsuits that only exist in aseprite for now

Honestly I'd love to see those in game, would love for a paramedic to pull up like trauma team haha

**Comment by Plum-but-gayer** [comment]
are there any plans for the 'Val' to get an enclosed helmet? I really love how the enclosed helmets look on the Archangel and Mur!

File: modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm [hunk 2]
```diff
@@ -2,21 +2,59 @@
...
+
+
+/datum/armament_entry/company_import/sol_defense/armor_hardened/emt_enclosed_helmet
+	item_type = /obj/item/clothing/head/helmet/toggleable/sf_hardened/emt
+
+/datum/armament_entry/company_import/sol_defense/armor_hardened/hardened_vest
+	item_type = /obj/item/clothing/suit/armor/sf_hardened
+
+/datum/armament_entry/company_import/sol_defense/armor_hardened/emt_hardened_vest
+	item_type = /obj/item/clothing/suit/armor/sf_hardened/emt
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm:46]: Why would I take the soft vests if the hardened vests cost the exact same amount of credits?

File: modular_skyrat/modules/specialist_armor/code/cargo_packs.dm [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	contains = list(
+		/obj/item/clothing/head/helmet/sf_peacekeeper/debranded = 3,
+		/obj/item/clothing/suit/armor/sf_peacekeeper/debranded = 3,
+	)
+
+/datum/supply_pack/security/armory/hardened_armor
+	name = "Hardened Armor Kit Crate"
+	crate_name = "hardened armor kit crate"
+	desc = "Contains three sets of SolFed-made hardened body armor and matching helmets."
+	access = ACCESS_MEDICAL
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/specialist_armor/code/cargo_packs.dm]: ... Medical?

File: modular_skyrat/modules/specialist_armor/code/cargo_packs.dm [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/datum/supply_pack/security/armory/soft_armor
+	name = "Soft Armor Kit Crate"
+	crate_name = "soft armor kit crate"
+	desc = "Contains three sets of SolFed-made soft body armor and matching helmets."
+	access = ACCESS_MEDICAL
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/specialist_armor/code/cargo_packs.dm]: ... Medical?

File: modular_skyrat/modules/specialist_armor/code/cargo_packs.dm [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		/obj/item/clothing/head/helmet/sf_peacekeeper/debranded = 3,
+		/obj/item/clothing/suit/armor/sf_peacekeeper/debranded = 3,
+	)
+
+/datum/supply_pack/security/armory/hardened_armor
+	name = "Hardened Armor Kit Crate"
+	crate_name = "hardened armor kit crate"
+	desc = "Contains three sets of SolFed-made hardened body armor and matching helmets."
+	access = ACCESS_MEDICAL
+	cost = CARGO_CRATE_VALUE * 5
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/specialist_armor/code/cargo_packs.dm:15]: This is the same exact cost as the soft armor vests, why would anyone go for a soft armor vest over a hardened one?

**Comment by Paxilmaniac** [comment]
> are there any plans for the 'Val' to get an enclosed helmet? I really love how the enclosed helmets look on the Archangel and Mur!

I could probably make a face shield or something for it

File: modular_skyrat/modules/specialist_armor/code/cargo_packs.dm [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+		/obj/item/clothing/head/helmet/toggleable/sf_hardened = 3,
+		/obj/item/clothing/suit/armor/sf_hardened = 3,
+	)
+
+/datum/supply_pack/security/armory/sacrificial_armor
+	name = "Sacrificial Armor Kit Crate"
+	crate_name = "sacrificial armor kit crate"
+	desc = "Contains three sets of SolFed-made sacrificial body armor and matching helmets."
+	access = ACCESS_MEDICAL
+	cost = CARGO_CRATE_VALUE * 5
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/specialist_armor/code/cargo_packs.dm]: Same comments here.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/specialist_armor/code/cargo_packs.dm (hunk 3 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/specialist_armor/code/cargo_packs.dm]: Copy and paste moment oops

File: modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm (hunk 2 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/armament_datums/sol_defense.dm:46]: The "soft" vests have significantly higher wound protection (twice the amount!)

File: modular_skyrat/modules/specialist_armor/code/cargo_packs.dm (hunk 5 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/specialist_armor/code/cargo_packs.dm:15]: They have significantly higher wound protection than the hardened vests do, twice as much as mentioned earlier.

File: modular_skyrat/modules/specialist_armor/code/cargo_packs.dm (hunk 6 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/specialist_armor/code/cargo_packs.dm]: And considering that the sacrificial armor physically breaks and falls off you without any ability to repair it, someone who's not into that kinda thing would choose one of the other armors.

**Review by GoldenAlpharex — COMMENTED** [review]
I'll give it a green light for a test-merge, we'll see how it goes.

**Comment by Samarai** [comment]
Sprites for the armors are missing/broken on digitigrade characters.

**Comment by Namwty** [comment]
Some helmets like 'VAL' sacrificial one do not respect snouts it seems, which look really odd ( snoutless folk who clearly are meant to have snout *shiver ).

Would be neat if that is looked into, but ye, otherwise kinda neat armor ( still waiting for that jumpsuit drip )

**Comment by Paxilmaniac** [comment]
> Some helmets like 'VAL' sacrificial one do not respect snouts it seems, which look really odd ( snoutless folk who clearly are meant to have snout *shiver ).
> 
> Would be neat if that is looked into, but ye, otherwise kinda neat armor ( still waiting for that jumpsuit drip )

Is this with or without the face shield installed.

File: modular_skyrat/modules/specialist_armor/code/sacrificial.dm [hunk 7]
```diff
@@ -71,11 +58,77 @@
...
+
+	if(!(istype(attacking_item, /obj/item/sacrificial_face_shield)))
+		return
+
+	add_face_shield(user, attacking_item)
+
+/obj/item/clothing/head/helmet/sf_sacrificial/AltClick(mob/user)
+	remove_face_shield(user)
+
+/obj/item/clothing/head/helmet/sf_sacrificial/proc/add_face_shield(mob/living/carbon/human/user, obj/shield_in_question)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/specialist_armor/code/sacrificial.dm:86]: Document this please!

File: modular_skyrat/modules/specialist_armor/code/sacrificial.dm [hunk 8]
```diff
@@ -71,11 +58,77 @@
...
+	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH | PEPPERPROOF
+
+	playsound(src, 'sound/items/modsuit/magnetic_harness.ogg', 50, TRUE)
+	face_shield = shield_in_question
+
+	icon_state = "bulletproof_glass"
+	worn_icon_state = icon_state
+	update_appearance()
+
+/obj/item/clothing/head/helmet/sf_sacrificial/proc/remove_face_shield(mob/living/carbon/human/user, break_it)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/specialist_armor/code/sacrificial.dm:103]: Document this please!

File: modular_skyrat/modules/specialist_armor/code/sacrificial.dm [hunk 9]
```diff
@@ -71,11 +58,77 @@
...
+
+	if((damage_amount > 20) && face_shield)
+		remove_face_shield(break_it = TRUE)
+
+/obj/item/clothing/head/helmet/sf_sacrificial/examine(mob/user)
+	. = ..()
+	if(face_shield)
+		. += span_notice("The <b>face shield</b> can be removed with <b>Right-Click</b>.")
+	else
+		. += span_notice("A <b>face shield</b> can be attached to it, if you had one.")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/specialist_armor/code/sacrificial.dm]:
```suggestion
		. += span_notice("A <b>face shield</b> can be attached to it.")
```

File: modular_skyrat/modules/specialist_armor/code/sacrificial.dm [hunk 10]
```diff
@@ -71,11 +58,77 @@
 	repairable_by = null // No being cheeky and keeping a pile of repair materials in your bag to fix it either
 	dog_fashion = null
 	flags_inv = null
+	supports_variations_flags = CLOTHING_SNOUTED_VARIATION_NO_NEW_ICON
+	/// Holds the faceshield for quick reference
+	var/obj/item/sacrificial_face_shield/face_shield
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/specialist_armor/code/sacrificial.dm:63]: Make sure to handle destroying this when you qdel the helmet.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Review by GoldenAlpharex — APPROVED** [review]
Should be good to go.

**Pull request merged**